spmd/tensor/device_mesh.py

```diff
@@ -391,3 +392,22 @@ def reduce_scatter_base(
 
             chunks = reduced_tensor.chunk(num_chunks, dim=shard_dim)
             return self.scatter(chunks, mesh_dim=mesh_dim)
+
+    def all_to_all(
+        self, input_tensor_list: List[torch.Tensor], mesh_dim: int = 0
+    ) -> List[torch.Tensor]:
+        # TODO: shall we make input tensors contiguous???
+        to_broadcast = [
+            CommTensor(tensor.contiguous()) for tensor in input_tensor_list
+        ]
+        dim_group = self._dim_groups[mesh_dim]
+        # all_to_all is not supported on 'gloo'
+        if self.backend() == "gloo":
+            raise RuntimeError(
```

Comment:
I know that all_to_all don't have a gloo backend, but we do need all_to_all like feature in CPU tests, could you add a fallback for all_to_all in gloo? i.e. In this case you can `all_gather` the tensor first and split the all_gathered tensor on the other tensor dimension, see a similar handling about reduce_scatter in Gloo [here](https://github.com/pytorch/tau/pull/461/files#diff-7011fb0a506b9cac502d4295ccb898f9a987fa42837fe0eda650143fcf2113dbR464) (where only nccl have reduce_scatter)

Reply:
I see. So I use `dist.scatter` and `dist.gather_all` instead of calling `dist.all_to_all` directly on CPU to perform the operation, right?

Reply:
Yes, that virtually looks like a transpose or what the all_to_all does